crates/egui_extras/src/loaders/gif_loader.rs

```diff
@@ -0,0 +1,139 @@
+use egui::{
+    ahash::HashMap,
+    load::{Bytes, BytesPoll, ImageLoadResult, ImageLoader, ImagePoll, LoadError, SizeHint},
+    mutex::Mutex,
+    ColorImage, ImageDataIdIndex,
+};
+use image::AnimationDecoder as _;
+use std::{io::Cursor, mem::size_of, sync::Arc, time::Duration};
+
+/// Array of Frames and the duration for how long each frame should be shown
+#[derive(Debug, Clone)]
+pub struct AnimatedImage {
+    frames: Vec<Arc<ColorImage>>,
+    delays: Arc<Vec<Duration>>,
+}
+
+impl AnimatedImage {
+    pub fn byte_len(&self) -> usize {
+        size_of::<Self>()
+            + self
+                .frames
+                .iter()
+                .map(|image| {
+                    image.pixels.len() * size_of::<egui::Color32>() + size_of::<Duration>()
+                })
+                .sum::<usize>()
+    }
+
+    /// Gets image at index
+    pub fn get_image(&self, index: usize) -> Arc<ColorImage> {
+        self.frames.get(index % self.frames.len()).cloned().unwrap()
```

Comment:
`.get(…).unwrap()` is just `[…]` with extra steps
```suggestion
        self.frames[index % self.frames.len()].clone()
```